cli/tests/test_next_prev_commands.rs

```diff
@@ -1112,6 +1112,405 @@ fn test_movement_edit_mode_false() {
     "###);
 }
 
+#[test]
+fn test_movement_target_revset() {
```

Comment:
Some of the added tests look redundant. What we need to test in this patch is `movement.edit = false` but the wc commit has children (i.e. currently editing), right?

Reply:
Good point. I got carried away and added test that would already be covered by other tests. D'oh!

I've pruned most of them now. In particular, I've remove `edit = true` and any `prev` related stanzas.

PTAL.

Reply:
You'll need to add tests in which `@` isn't a head. Try run the added tests against the previous commit to see if they fail.

Reply:
The only case I've found which isn't covered existing tests is the case where all the following hold.

* edit=false
* wc is not a head
* no conflict
* offset > 1 (I still added a test, but it's not really interesting)

I've added a test for that and confirm that it fails without the code change in this PR. See #4484 

The conflict case doesn't have this problem in trunk; when `edit=false` since we end up taking the `root()` conflicted descendants and then removing the wc.

Let me know if there's any other edge case that would make sense to explicitly test for.

Reply:
Thanks. I think we could add test in which both sides have conflicts 
(you'll get prompt without this patch), but that's not so important. I think the added test is good enough.

Reply:
Thanks. Resolving now.